internal/processors/httpinputserver/http_input_server.go

```diff
@@ -1,23 +1,125 @@
+
+var field = service.NewObjectField("jwt",
+	service.NewStringField(tokenExchangeIssuer).Description("token exchange service"),
+	service.NewStringField(tokenExchangeKeySetURL).Description("provides public keys for jwt signature validation"),
+)
 
 func init() {
-	io.RegisterCustomHTTPServerInput("dimo_http_server", CertRoutingMiddlewarefunc)
+	io.RegisterCustomHTTPServerInput("dimo_http_connection_server", CertRoutingMiddlewareConstructor, nil)
+	io.RegisterCustomHTTPServerInput("dimo_http_attestation_server", AttestationMiddlewareConstructor, field)
+}
+
+func CertRoutingMiddlewareConstructor(*service.ParsedConfig) (func(*http.Request) (map[string]any, error), error) {
+	return CertRoutingMiddlewarefunc, nil
 }
 
 func CertRoutingMiddlewarefunc(r *http.Request) (map[string]any, error) {
 	retMeta := map[string]any{}
 	verifiedChains := r.TLS.VerifiedChains
 	for _, certChain := range verifiedChains {
 		firstCert := certChain[0]
-		retMeta[DIMOConnectionIdKey] = firstCert.Subject.CommonName
+		retMeta[DIMOCloudEventSource] = firstCert.Subject.CommonName
+		retMeta[processors.MessageContentKey] = ConnectionContent
 	}
 	return retMeta, nil
 }
+
+func AttestationMiddlewareConstructor(conf *service.ParsedConfig) (func(*http.Request) (map[string]any, error), error) {
+	return attestationMiddleware(conf)
+}
+
+func attestationMiddleware(conf *service.ParsedConfig) (func(*http.Request) (map[string]any, error), error) {
+	subConf := conf.Namespace("jwt")
+	issuer, err := subConf.FieldString(tokenExchangeIssuer)
+	if err != nil {
+		return nil, err
+	}
+
+	jwksURI, err := subConf.FieldString(tokenExchangeKeySetURL)
+	if err != nil {
+		return nil, err
+	}
+
+	issuerURL, err := url.Parse(issuer)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse issuer URL: %w", err)
+	}
+
+	opts := []any{}
+	if jwksURI != "" {
+		keysURI, err := url.Parse(jwksURI)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse jwksURI: %w", err)
+		}
+		opts = append(opts, jwks.WithCustomJWKSURI(keysURI))
+	}
+	provider := jwks.NewCachingProvider(issuerURL, 1*time.Minute, opts...)
+	// Set up the validator.
+	jwtValidator, err := validator.New(
+		provider.KeyFunc,
+		validator.RS256,
+		issuerURL.String(),
+		[]string{"dimo.zone"},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create validator: %w", err)
+	}
+
+	return func(r *http.Request) (map[string]any, error) {
+		retMeta := map[string]any{}
+		authStr := r.Header.Get("Authorization")
+		tokenStr := strings.TrimSpace(strings.Replace(authStr, "Bearer ", "", 1))
+		tkn, err := jwtValidator.ValidateToken(r.Context(), tokenStr)
+		if err != nil {
+			return retMeta, fmt.Errorf("invalid token string: %w", err)
+		}
+
+		token, ok := tkn.(jwt.Token)
+		if !ok {
+			return retMeta, fmt.Errorf("unexpted token type")
+		}
+
+		claims := token.Claims.(jwt.MapClaims)
+		ethAddr, exists := claims["ethereum_address"].(string)
+		if exists {
+			return retMeta, fmt.Errorf("no ethereum address in token")
```

Comment:
thought: Is it dumb for me to say "hey do `errors.New` if there aren't any args"?